apps/bplan/serializers.py

```diff
@@ -104,12 +102,35 @@ def _get_absolute_url(self, bplan):
         return url
 
     def _download_image_from_url(self, url):
-        parsed_url = urlparse(url)
-        file_path = os.path.join(PROJECT_IMAGE_DIR,
-                                 os.path.basename(parsed_url.path))
-        file_name = os.path.join(settings.MEDIA_ROOT, file_path)
-        file_dir = os.path.dirname(file_name)
-        os.makedirs(file_dir, exist_ok=True)
-
-        request.urlretrieve(url, file_name)
+        try:
+            parsed_url = urlparse(url)
+            file_path = os.path.join(PROJECT_IMAGE_DIR,
+                                     os.path.basename(parsed_url.path))
+            file_name = os.path.join(settings.MEDIA_ROOT, file_path)
+            file_dir = os.path.dirname(file_name)
+            os.makedirs(file_dir, exist_ok=True)
+
+            request.urlretrieve(url, file_name)
+        except:
+            self._remove_image_if_exists(file_name)
+            raise serializers.ValidationError(
+                'Failed to download image {}'.format(url))
+        self._validate_image(file_name)
```

Comment:
why are you validating the image outside of the try block?
i suggest to remove the try/except block from the _validate_image function and just move the call in the try block